### DISCUSSION TITLE
Welcome to EEG-ATCNet Discussions!

### DISCUSSION THREAD

**Discussion by Altaheri** [Altaheri/EEG-ATCNet#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by marianicolae24** [comment]
Hello! I wonder why you define the model and compile it for each training, and also why  you add the 'for train in nr_trainings' instead of just using the epochs.

**Reply by Altaheri** [reply]
Each "train" is a separate run (repetition) and the model must be initialized with random weights by defining and compiling a new model. You can train the model on one run and get the evaluation accuracy. Otherwise, you can train the model in several runs (repetitions) and get the average and best accuracy among the runs.

The loop in the following line is for separate and distinct training and evaluation procedures:
```
# Iteration over multiple runs
for train in range(n_train):
```
In Keras, the iteration over the number of epochs is hidden. They are hidden in the following line:
```
history = model.fit(X_train, y_train_onehot, validation_data=(X_test, y_test_onehot), epochs=epochs, batch_size=batch_size, callbacks=callbacks, verbose=0)
```
